station_distance.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+require 'open-uri'
+
+def filescrapper(url)
+  csv_text = open(url)
+  CSV.parse(csv_text, headers: true, col_sep: ';')
+end
+
+def to_rad(angle)
+  angle.to_f * 0.017453292519943295
+end
+
+def distance_calc(coord)
+  a1 = Math.sin(coord[3] - coord[1])
+  a2 = Math.cos(coord[3]) * Math.cos(coord[1])
+  a3 = Math.sin(coord[2] - coord[0])
+  a = a1**2 + a2 * a3**2
+  b = Math.sqrt(a) / Math.sqrt(1 - a)
+  c = Math.atan(b)
+  (6371 * c).to_i
+end
+
+def station_user_selector(station_array)
+  station = []
+  p 'please select departure station id'
+  station << station_array[gets.chomp.to_i - 1]
+  p 'please select arrival station id'
+  station << station_array[gets.chomp.to_i - 1]
+end
+
+def station_select(station_list)
+  i = 0
+  station_array = []
+  station_list.each do |row|
+    i += 1
+    station_array << [i, row[1], row[5], row[6]]
+    break if i >= 50
+  end
+  station_array
+end
+
+def coordinates_buider(station)
+  long1 = to_rad(station[0][3].to_f)
+  lat1 = to_rad(station[0][2].to_f)
+  long2 = to_rad(station[1][3].to_f)
+  lat2 = to_rad(station[1][2].to_f)
+  [long1, lat1, long2, lat2]
+end
+
+def distance_interface
+  p 'Downloading list of stations, please wait...'
+  # station_list = filescrapper('stations.csv')
+  station_list = filescrapper('https://raw.githubusercontent.com/trainline-eu/stations/master/stations.csv')
+  station_array = station_select(station_list)
+  station_array.each { |st| p "#{st[0]} - #{st[1]}" }
```

Comment:
also here, perhaps using a hash with clear keys would help understand what we are displaying without having to dig into the `select_stations` method

Reply:
Right, dealing with hashes is more convenient.